validations/src/validations.rs

```diff
@@ -1442,8 +1444,9 @@ pub fn validate_block_transactions(
     // TODO: replace for loop with a try_fold
     // Validate value transfer transactions in a block
     let mut vt_mt = ProgressiveMerkleTree::sha256();
+    let mut vt_weight = 0;
     for transaction in &block.txns.value_transfer_txns {
-        let (inputs, outputs, fee) = if is_genesis {
+        let (inputs, outputs, fee, weight) = if is_genesis {
```

Comment:
I guess that the genesis block can be unlimited in size? Let's add a test, just in case...

Reply:
Yep, good catch!

Reply:
done!